content/en/localstack/init-hooks.md

```diff
@@ -0,0 +1,146 @@
+        └── start.d          <-- executed when localstack starts up
+```
+
+In these directories, you can put either executable shell scripts or python programs, which will be executed from within a Python process.
+All except `boot.d` will be run in the same Python interpreter as LocalStack, which gives additional ways of configuring/extending LocalStack.
+
+Currently, known script extensions are `.sh`, and `.py`.
+Shell scripts have to be executable.
+
+A script can be in one of four states: `UNKNOWN`, `RUNNING`, `SUCCESSFUL`, `ERROR`.
+Scripts are by default in the `UNKNOWN` state once they have been discovered.
+The remaining states should be self-explanatory.
+
+### Execution order and script failures
+
+Scripts are sorted and executed in alphanumerical order.
+If an init script fails, the remaining scripts will still be executed in order.
+A script is considered in `ERROR` state if it is a shell script and returns with a non-zero exit code, or if a Python script raises an exception during its execution.
+
+
+## Status endpoint
+
+There is an additional endpoint in `localhost:4566/_localstack/init` which returns the state of the initialization procedure.
+Boot scripts are currently always in the unknown state, since they are executed outside the localstack process and we don't know whether they have been successfully executed or not.
+
+```bash
+curl -s localhost:4566/_localstack/init | jq .
+```
+```json
+{
+  "completed": {
+    "BOOT": false,
+    "START": true,
+    "READY": true,
+    "SHUTDOWN": false
+  },
+  "scripts": [
+    {
+      "stage": "BOOT",
+      "name": "booting.sh",
+      "state": "UNKNOWN"
+    },
+    {
+      "stage": "READY",
+      "name": "pre_seed.py",
+      "state": "SUCCESSFUL"
+    }
+  ]
+}
+```
+
+### Query individual stages
+
+You can also query a specific stage:
+```bash
+curl -s localhost:4566/_localstack/init/ready | jq .
+```
+```json
+{
+  "completed": true,
+  "scripts": [
+    {
+      "stage": "READY",
+      "name": "pre_seed.py",
+      "state": "OK"
+    }
+  ]
+}
+```
+
+To check whether a given stage has been completed you can now run, for example:
+
+```bash
+curl -s localhost:4566/_localstack/init/ready | jq .completed
+```
+which returns either `true` or `false`.
+
+
+## Usage example
+
+A common use case for init hooks is pre-seeding LocalStack with custom state.
+If you have more complex state, [Cloud Pods]({{< ref `cloud-pods` >}}) may be a good option to look into!
+But for simple state, for example if you want to have a certain S3 bucket or DynamoDB table created when starting LocalStack, init hooks can be very useful.
+
+To execute aws cli commands when LocalStack becomes ready,
+simply create a script `init-aws.sh` and mount it into `/etc/localstack/init/ready.d/`.
+Make sure the script is executable: run `chmod +x init-aws.sh` on the file first.
+You can use anything available inside the container, including `awslocal`:
```

Comment:
Kinda dangerous TBH. Not in the sense of security but because technically everything we have in the container is part of our public API now and thus subject to semver :grimacing: 

Maybe we should limit this to a few but more strictly defined tools?

Reply:
Good point!